login.go

```diff
@@ -97,6 +101,10 @@ func getDeviceCode(ctx context.Context, client *ConductoroneAPI, clientID string
 		return nil, "", err
 	}
 
+	if resp.StatusCode >= 300 {
```

Comment:
the expected status codes should be known yeah? Can we explicitly say `if resp.StatusCode != http.StatusOk`? or something similar?